crates/hir_ty/src/lib.rs

```diff
@@ -1535,16 +1548,18 @@ mod tests {
     fn infer_function_param() {
         check_in_root_file(
             r#"
-                fn aaa(x: int, y: string) {
-                    let a = x
-                    let b = y
+                fn aaa(x: int, y: string, z: mut bool) {
+                    let a = x;
+                    let b = y;
+                    let c = z;
```

Comment:
**Performance Issue**: Adding a mutable boolean 'z' in function parameters could lead to side effects if 'z' is altered within the function, potentially leading to harder to track bugs and performance issues if 'z' is shared across threads. <br> **Fix**: If 'z' does not need to be mutable across different calls or within the function, consider removing the 'mut' keyword or use a local copy. <br> **Code Suggestion**: 
 ```
 -                fn aaa(x: int, y: string, z: mut bool) {
 +                fn aaa(x: int, y: string, z: bool) {
 ```